gen/AArch64Compile_gen.ml

```diff
@@ -1725,7 +1725,20 @@ module Make(Cfg:Config) : XXXCompile_gen.S =
 
     let get_tagged_loc e = add_tag (as_data e.C.loc) e.C.tag
 
-    let emit_access st p init e = 
+    let add_label_to_instructions e cs =
+      match e.C.check_fault with
+      | Some (label_name, _) ->
+        (* Always label the last instruction,
+           which should be the actual load or store. *)
+        let length = List.length cs in
+        List.mapi
+        ( fun index instr ->
+            if index = (length - 1) then
+                Label(label_name, instr)
+            else instr ) cs
```

Comment:
Some may find the following code to be more straightforward:
```suggestion
          let rec do_rec = function
            | [] -> assert false
            | [instr] -> [Label (label_name,instr)]
            | instr::rem -> instr::do_rec rem in
          do_rec cs
```

Reply:
Will do. I will probably return [] on [] case.